imap_processing/ultra/l1c/spacecraft_pset.py

```diff
@@ -63,22 +66,26 @@ def calculate_spacecraft_pset(
     """
     pset_dict: dict[str, np.ndarray] = {}
     sensor = parse_filename_like(name)["sensor"][0:2]
+    indices = np.where(de_dataset["species"].values == species_id)[0]
```

Comment:
Add a quick comment about this being the place where you are only keeping this species_id.